### ISSUE TITLE
Prediction Probability of the T5 Model

### ISSUE THREAD

**Issue by abedkhooli** [ThilinaRajapakse/simpletransformers#878]
I am trying T5 (and mT5) with success except for the prediction probability and I have seen the related issues for other models (Softmax on raw output) but the [predict of T5](https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/t5/t5_model.py#L781) does not output the raw part. Is there a way to get the prediction probability in T5?

**Comment by ThilinaRajapakse** [comment]
T5 and mT5 are generative models (unlike BERT). You can't get a prediction probability like that for T5.

**Comment by abedkhooli** [comment]
Say I have a binary classification task (positive, negative) and no other tasks. The model (T5, mT5) generates an output (either or) - doesn't this have some probability associated? Or is it based on some loss that can't be converted to probability?

**Comment by ThilinaRajapakse** [comment]
It will generate a sequence of language model logits. They can be converted to probabilities, but they won't be probabilities for a binary classification task. It'll be a probability distribution over the entire vocabulary of the model.

The task is irrelevant as T5 is a Sequence-to-Sequence model. The generated sequence may correspond to a binary label but the model (and its outputs) won't reflect that.

**Comment by dipanjannC** [comment]
So, is there any other way to solve this problem. As we might create some additional labels using the prediction probability of the labels predicted?

**Comment by dzombiee** [comment]
You can get output_scores by passing it as True in generate method, then run a softmax over it, you'll get probability for each token predicted. You can use these some way by taking mean/weighted mean etc, but getting one probability per output is not possible in T5.

Hope this helps :)

**Comment by WeiShi-9** [comment]
> Is it possible to use "loss" as the prediction score of each given target sequence? As I know, loss is calculated based on the probabilities of each target token.

So how to get loss from T5 model?
